server/jetstream_cluster.go

```diff
@@ -3848,8 +3848,29 @@ func (s *Server) jsClusteredStreamListRequest(acc *Account, ci *ClientInfo, offs
 
 	var streams []*streamAssignment
 	for _, sa := range cc.streams[acc.Name] {
-		streams = append(streams, sa)
+		if IsNatsErr(sa.err, JSClusterNotAssignedErr) {
+			continue
+		}
+
+		if filter != _EMPTY_ {
+			// These could not have subjects auto-filled in since they are raw and unprocessed.
+			if len(sa.Config.Subjects) == 0 {
+				if SubjectsCollide(filter, sa.Config.Name) {
```

Comment:
Should it be collide or do we want the filter to be equal or a subset of the subjects attached to the stream?

Reply:
Don’t know tbh. I copied logic from stream names they should behave the same. So this is whatever names does :)

Reply:
ok so this is from names filter code, so agree using same thing makes sense here.